gc/src/lib.rs

```diff
@@ -30,17 +32,13 @@ pub use gc::force_collect;
 ///
 /// See the [module level documentation](./) for more details.
 pub struct Gc<T: Trace + ?Sized + 'static> {
-    // XXX We can probably take advantage of alignment to store this
-    root: Cell<bool>,
-    _ptr: NonZero<*mut GcBox<T>>,
+    _ptr: UnsafeCell<NonZero<*mut GcBox<T>>>,
```

Comment:
I'm pretty sure that it's fine, but I'd also be OK with dropping it until we find someone who would know better. The impression I have about NonZero, is that it is used as a marker trait for Zeroable types which guarantees that the zero value of the type may be used as a marker value for variants. I don't think that this information is really communicated to llvm that much, and also none of the computations which I've seen in the patch I believe will ever cause problems with setting something to zero.